### PR TITLE
Pricing page: add new header

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

This PR ports the new [Jetpack.com](https://jetpack.com/) header to the Jetpack pricing page.

### Implementation notes

The implementation differs slightly from Jetpack.com:
- The search features is absent
- So is the Upgrade/Get started button
- Because of the above, the "More" functionality that wraps  in a menu overflowing items on small screens was not implemented either
- CSS and JS code is copied from Jetpack.com. It's not optimal but it made the implementation way faster.
- The header is not fixed in the pricing page

### Testing instructions

- Download the PR and run cloud
- Visit `/pricing`
- Check that the header is similar to the Jetpack.com header (keeping in mind the aforementioned notes)
- Check on desktop/mobile, authenticated/unauthenticated
- In a different locale than English, check that links bring you to Jetpack.com in the correct language

### Screenshots
Screen Shot 2022-02-15 at 2 08 20 PM
Screen Shot 2022-02-15 at 2 08 27 PM

### CODE REVIEW

File: client/jetpack-cloud/sections/pricing/jpcom-masterbar/index.tsx [hunk 1]
```diff
@@ -1,282 +1,285 @@
...
 					{
-						category: {
-							label: translate( 'Security' ),
-							description: translate( 'Protect your site' ),
-							href: '/features/security/',
-							icon: securityIcon,
-						},
+						label: translate( 'Security' ),
+						tagline: translate( 'Protect your site' ),
+						href: `${ JETPACK_COM_BASE_URL }/features/security/`,
```
Comment (travisw) [inline client/jetpack-cloud/sections/pricing/jpcom-masterbar/index.tsx:51]: ~~Do you think keeping these URLs as they are without the added base URL might make it easier to test? For example, I'm currently using a jetpack cloud URL (eg. `https://container-goofy-brahmagupta.calypso.live/pricing`) and when navigating these links I'd expect it to stay on the `container-goofy-brahmagupta.calypso.live` site but instead it always goes to `jetpack.com`.~~

I just tested this on another branch and it behaves the same way so this doesn't seem to be a change with this PR. 👍

File: client/jetpack-cloud/sections/pricing/jpcom-masterbar/index.tsx [hunk 2]
```diff
@@ -1,282 +1,285 @@
...
-		},
-		[ toggleSubmenu ]
-	);
 	const onLinkClick = useCallback( ( e ) => {
 		recordTracksEvent( 'calypso_jetpack_nav_item_click', {
 			nav_item: e.currentTarget.getAttribute( 'href' )?.replace( JETPACK_COM_BASE_URL, '' ),
 		} );
 	}, [] );
 
+	const getLocalizedLink = useCallback(
```
Comment (travisw) [inline client/jetpack-cloud/sections/pricing/jpcom-masterbar/index.tsx]: Do we not already have some shared functions to handle things like this?
Maybe [`localizeUrl`](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-utils/src/localize-url.tsx#L153) or something similar?

**Review by travisw — COMMENTED** [review]
It works well. I left a few relatively minor comments.

---

The "Manage your sites" pop-up is running off the page for me on a mid-sized screen.
Screen Shot 2022-02-16 at 2 55 27 PM

---

I'm not sure if it matters but we're no longer showing an indication that we're on the pricing page by underlining the "pricing" nav item. I guess that goes for the other nav items too.
Here's how it shows on the current/old /pricing page:
Screen Shot 2022-02-16 at 3 04 01 PM

---

I didn't look that closely at the files that were imported from jetpack.com

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7144432

Thank you @monsieur-z for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7144432).

**Pull request merged**